src/app.rs

```diff
@@ -634,6 +634,8 @@ impl App {
   }
 
   pub fn pop_navigation_stack(&mut self) -> Option<Route> {
+    self.navigation_stack.dedup_by(|a, b| a.id == b.id);
```

Comment:
I think it would be just a bit nicer to do this in `push_navigation_stack`, assuming that that is the only place where the stack is stack is grown. In that case, it could possibly be just an `if` statement checking whether the current id is the same as the id at the top of the stack.

Reply:
thanks for looking at this 👍 moved to `push_navigation_stack` and changed to if statement.